ios/Classes/OfflineRegionData.swift

```diff
@@ -0,0 +1,101 @@
+//
+//  OfflineRegionData.swift
+//  location
+//
+//  Created by Patryk on 02/06/2020.
+//
+
+import Foundation
+import Mapbox
+
+class OfflineRegionData {
+    var id: Int
```

Comment:
I found the integer `id` here to be a bit confusing in its behavior. I assumed uniqueness would be guaranteed, but it's not, I can add many regions with id `0`. If I ask for `0` to be deleted, the first of these is deleted.

Knowing that uniqueness is not enforced, rather than maintain a counter in the application code I would use a randomly generated unique ID, normally a UUID. But since a UUID can't be stored as an int (string is easiest), I would generate a random int value and provide that to the framework.

So that's how I can implement if you don't make changes, but I suggest managing this issue within the library. Rather than requiring that the app specify an `id` inside an `OfflineRegion` when calling `downloadOfflineRegionStream` and `downloadOfflineRegion`, you can generate a UUID at that point and use it going forward. This would require changing the `id` type to a `string`, and it would require a different interface to the download methods which requires all of the fields from `OfflineRegion` except `id`. I don't think providing `id` should be supported even as an option unless the library is going to enforce uniqueness, and that's much harder than generating its own unique ids every time.

Reply:
The way the id is implemented is for backend integrity purpose. As the mapbox generate its own id for the downloading region. The one that should be provided is to make possible to proper identify region at the moment when we added it because there is no way to know it before by using only builded-in mapbox solution. So from mapbox side the item will have unique identifier. As for me i am not sure think that library should not handle situations like this.

Reply:
The solution to this problem for symbols/circles/lines is there is a `SymbolOptions` class with all options the developer can set, and a `Symbol` class returned by `addSymbol()` which contains the generated, unique id and the symbolOptions. I think a similar approach would work well for offline regions. You pass an `OfflineRegionOptions` object with all options except for the id to `downloadRegion()` and it returns an `OfflineRegion` object with the generated id and the offlineRegionOptions.